includes/Embed_Block.php

```diff
@@ -0,0 +1,171 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+/**
+ * Embed block class.
+ */
+class Embed_Block {
+	/**
+	 * Script handle.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_HANDLE = 'web-stories-embed-block';
+
+	/**
+	 * Style handle.
+	 *
+	 * @var string
+	 */
+	const STYLE_HANDLE = 'web-stories-embed-block';
+
+	/**
+	 * Initializes the Web Stories embed block.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		wp_register_script( 'amp-story-player', 'https://cdn.ampproject.org/amp-story-player-v0.js', [], 'v0', false );
+		wp_register_style( 'amp-story-player', 'https://cdn.ampproject.org/amp-story-player-v0.css', [], 'v0' );
+
+		$asset_file   = WEBSTORIES_PLUGIN_DIR_PATH . 'assets/js/' . self::SCRIPT_HANDLE . '.asset.php';
+		$asset        = is_readable( $asset_file ) ? require $asset_file : [];
+		$dependencies = isset( $asset['dependencies'] ) ? $asset['dependencies'] : [];
+		$version      = isset( $asset['version'] ) ? $asset['version'] : WEBSTORIES_VERSION;
+
+		$dependencies[] = 'amp-story-player';
+
+		wp_register_script(
+			self::SCRIPT_HANDLE,
+			WEBSTORIES_PLUGIN_DIR_URL . 'assets/js/' . self::SCRIPT_HANDLE . '.js',
+			$dependencies,
+			$version,
+			false
+		);
+
+		wp_register_style(
+			self::STYLE_HANDLE,
+			WEBSTORIES_PLUGIN_DIR_URL . 'assets/css/' . self::STYLE_HANDLE . '.css',
+			[ 'amp-story-player' ],
+			$version
+		);
+
+		// todo: use register_block_type_from_metadata() once generally available.
+
+		// Note: does not use 'script' and 'style' args, and instead uses 'render_callback'
+		// to enqueue these assets only when needed.
+		register_block_type(
+			'web-stories/embed',
+			[
+				'render_callback' => [ $this, 'render_block' ],
+				'editor_script'   => self::SCRIPT_HANDLE,
+				'editor_style'    => self::STYLE_HANDLE,
+			]
+		);
+
+		add_filter( 'wp_kses_allowed_html', [ __CLASS__, 'filter_kses_allowed_html' ], 10, 2 );
+	}
+
+	/**
+	 * Filter the allowed tags for KSES to allow for amp-story children.
+	 *
+	 * @param array|string $allowed_tags Allowed tags.
+	 *
+	 * @return array|string Allowed tags.
+	 */
+	public static function filter_kses_allowed_html( $allowed_tags ) {
+		if ( ! is_array( $allowed_tags ) ) {
+			return $allowed_tags;
+		}
+
+		$story_player_components = [
+			'amp-story-player' => [],
+		];
+
+		$allowed_tags = array_merge( $allowed_tags, $story_player_components );
+
+		return $allowed_tags;
+	}
+
+	/**
+	 * Renders the block type output for given attributes.
+	 *
+	 * @param array  $attributes Block attributes.
+	 * @param string $content    Block content.
+	 *
+	 * @return string Rendered block type output.
+	 */
+	public function render_block( array $attributes, $content ) {
+		$defaults = [
```

Comment:
These could also be set in `register_block_type` under `attributes` with types and defaults perhaps?

Reply:
Totally 👍